src/librustc_mir/transform/const_prop.rs

```diff
@@ -988,8 +1023,61 @@ impl<'mir, 'tcx> MutVisitor<'tcx> for ConstPropagator<'mir, 'tcx> {
             | TerminatorKind::GeneratorDrop
             | TerminatorKind::FalseEdges { .. }
             | TerminatorKind::FalseUnwind { .. } => {}
-            //FIXME(wesleywiser) Call does have Operands that could be const-propagated
-            TerminatorKind::Call { .. } => {}
+            TerminatorKind::Call { ref mut args, .. } => {
+                // FIXME(felix91gr) Finish this
+                for opr in args {
+                    // If it isn't a place, then it's a constant,
+                    // and thus we don't need to propagate towards it.
+                    trace!("Checking if arg {:?} is not a constant", &opr);
+                    if let Some(place) = opr.place() {
+                        // No constants from here on
+                        trace!("Obtained place: {:?}", &place);
+                        trace!("Checking if arg {:?} has no projections", &opr);
+                        if let Some(local) = place.as_local() {
+                            // No projections from here on
+                            trace!("Place has no projections. Local is {:?}", &local);
+                            let place_evaluated = self.eval_place(place);
+                            trace!("Place is evaluated  as {:?}", place_evaluated);
+                            if let Some(op_ty) = place_evaluated {
+                                trace!("OpTy of the place: {:?}", &op_ty);
+                                use std::ops::Deref;
+                                if let interpret::Operand::Immediate(immediate) = op_ty.deref() {
+                                    // Only Immediate from here on
+                                    // FIXME(felix91gr): This only works for Immediate. Investigate
+                                    // if it could be made to work for Indirect as well.
+                                    trace!("Immediate of the OpTy: {:?}", &immediate);
+                                    if let interpret::Immediate::Scalar(scalar_mu) = immediate {
+                                        // FIXME(felix91gr): This only works for Scalar
+                                        // Could probably be expanded for Scalar Tuples (inside the Immediate)
+                                        trace!("ScalarMaybeUndef of the OpTy: {:?}", &scalar_mu);
+                                        if let ScalarMaybeUndef::Scalar(scalar) = scalar_mu {
+                                            trace!("Scalar of the ScalarMU: {:?}", &scalar);
+                                            let place_type = op_ty.layout.ty;
```

Comment:
I think the `Ty` we want to use is actually the one from `opr.ty()`

Reply:
This was quite useful :)